wrappers/android/examples/sensor/src/main/java/com/intel/realsense/sensor/MainActivity.java

```diff
@@ -203,14 +210,40 @@ private void configAndStart() throws Exception {
                         if (w == 640 && fps == 30 && (sf.compareTo(StreamFormat.Z16) == 0)) {
                             Log.d(TAG, "depth stream: " + index + ":" + st.name() + ":" + sf.name() + ":" + w + "x" + h + "@" + fps + "HZ");
 
-                            sp = sp2;
-                            break;
+                            depth_profile = sp2;
+                            depth_profile_found = true;
+                        }
+                    }
+                }
+                if (!ir_profile_found && sp2.getType().compareTo(StreamType.INFRARED) == 0 && sp2.getIndex() == 1 ) {
+
+                    if (sp2.is(Extension.VIDEO_PROFILE)) {
+                        VideoStreamProfile video_stream_profile = sp2.as(Extension.VIDEO_PROFILE);
+
+                        // After using the "as" method we can use the new data type
+                        //  for additional operations:
+                        StreamFormat sf = video_stream_profile.getFormat();
+                        int index = sp2.getIndex();
+                        StreamType st = sp2.getType();
+                        int w = video_stream_profile.getWidth();
+                        int h = video_stream_profile.getHeight();
+                        int fps = video_stream_profile.getFrameRate();
+
+                        if (w == 640 && fps == 30 && (sf.compareTo(StreamFormat.Y8) == 0)) {
+                            Log.d(TAG, "ir stream: " + index + ":" + st.name() + ":" + sf.name() + ":" + w + "x" + h + "@" + fps + "HZ");
+
+                            ir_profile = sp2;
+                            ir_profile_found = true;
                         }
                     }
                 }
             }
 
```

Comment:
add exception if profiles were not found - or if only one was found, play only this one, and add a toast message on the missing profile